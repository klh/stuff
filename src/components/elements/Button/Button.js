import React from 'react';
import PropTypes from 'prop-types';
import {Button as DemoButton} from 'react-bootstrap';
import './Button.css'

/**
 * Button Stateless Element
 *
 * Demo Button Created to show how i would typically structure stateless components
 */

//Functional Component
const Button = React.memo((props) => (
    <DemoButton bsStyle={props.type}>{props.children}</DemoButton>
));

//PropTypes
Button.propTypes = {
    children: PropTypes.node,
    type: PropTypes.oneOf(['default', 'primary', 'dashed', 'danger'])
};

// Same approach for defaultProps too
Button.defaultProps = {
    children: 'Default Button Text',
    type: 'primary'
};

//etc

export default Button;
