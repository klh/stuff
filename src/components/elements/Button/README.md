# Button readme

### Description of methods etc.

lorem ipsum dolor sti amet.

### Code examples of useage

```
 <button />

```
