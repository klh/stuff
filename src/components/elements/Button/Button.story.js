import React from 'react';
import {storiesOf} from '@storybook/react';
import {action} from '@storybook/addon-actions';
import withDocs from 'storybook-readme/with-docs';
import ButtonReadme from './README.md';
import Button from './';

storiesOf('Button', module)
  .add(
    'Primary',
    withDocs(ButtonReadme, () => (
      <Button type="primary" onClick={action('clicked')}>
        Primary
      </Button>
    ))
  )
  .add(
    'Dashed',
    withDocs(ButtonReadme, () => (
      <Button type="dashed" onClick={action('clicked')}/>
    ))
  )
  .add(
    'Danger',
    withDocs(ButtonReadme, () => (
      <Button type="danger" onClick={action('clicked')}>
        Danger
      </Button>
    ))
  )
  .add(
    'default',
    withDocs(ButtonReadme, () => (
      <Button onClick={action('clicked')}>default</Button>
    ))
  );
