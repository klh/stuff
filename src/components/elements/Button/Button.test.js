import React from 'react';
import {render} from 'enzyme';

import Button from './';

describe('<Button />', () => {

    it('Snapshot mathes', () => {
        const ButtonRender = render(<Button>snapshot</Button>);
        expect(ButtonRender).toMatchSnapshot();
    });

    /*
    // Jest/ Enzyme does not yet support react.memo - but it'll be released in a week or so, then it would be good to test like so:

    it('No Children should match Default Text', () => {
        const defText = 'Default Button Text';
        const ButtonRender = mount(<Button/>);
        //expect(ButtonRender).toHaveText(defText);
        console.log(ButtonRender.text())
    });

    */

});

