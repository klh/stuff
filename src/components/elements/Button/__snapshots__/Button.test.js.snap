// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Snapshot mathes 1`] = `
<button
  class="btn btn-primary"
  type="button"
>
  snapshot
</button>
`;
